geopandas/tools/hilbert_curve.py

```diff
@@ -0,0 +1,172 @@
+import pandas as pd
+import numpy as np
+
+
+def _hilbert_distance(gdf, total_bounds, p):
+    """
+    Calculate hilbert distance for a GeoDataFrame
+    int coordinates
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+
+    total_bounds : Total bounds of geometries - array
+
+    p : The number of iterations used in constructing the Hilbert curve
```

Comment:
Elsewhere (GEOS, [Wikipedia](https://en.wikipedia.org/wiki/Hilbert_curve)) refers to this as `level` instead.  Suggest using `level` here.

Reply:
We also wanted to rename `p` in dask-geopandas. We thought about `precision` there but `level` could be fine as well. I guess that works for all options (Hilbert, Morton, Geohash?).

Reply:
Yes, for me `level` is fine as well.